Runtime/Scripts/Utils/ObserverUtils.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Linq;
+
+namespace Unity.MeshSync {
+    /// <summary>
+    /// Class that facilitates the automatic unsubscription of observers through the disposal pattern
+    /// </summary>
+    /// <typeparam name="T">Observer Type</typeparam>
+    /// <typeparam name="U">Observer data struct</typeparam>
+    internal sealed class Unsubscriber<T, U> : IDisposable where T : class, IObserver<U> {
+        
+        private bool disposed;
+        private IList<T> observers;
+        private T observer;
+
+        private Unsubscriber(IList<T> observers, T observer) {
+            this.observers = observers;
+            this.observer = observer;
+            disposed = false;
+        }
+
+        /// <summary>
+        /// Create an unsubscriber
+        /// </summary>
+        /// <remarks>Expects to be called after subscription</remarks>
+        /// <param name="observers">List of subscribed subscribers</param>
+        /// <param name="observer">Subscriber to unsubscribe on dispose</param>
+        /// <returns></returns>
+        public static Unsubscriber<T, U> Create(IList<T> observers, T observer) {
```

Comment:
It seems that the return value of `Unsubscriber.Create` is never used, as `MeshSyncObserverStart.update()` the only caller and it doesn't store this return value.
Do we need this ?

Reply:
This is a standard pattern (part of the larger Observer pattern) that allows the caller to unsubscribe using the disposal mechanism in .NET. As it stands right now the callers of this are static and thus can't implement IDispose, but there is value to have this mechanism in our codebase as the observer pattern is commonly used and can be powerful.

Reply:
I guess, more generically, then why are we using the Observer pattern here? 

If we're not implementing `OnNext` and `OnError` functionality, we're never actually using the unsubscriber logic because the callers of this are static, and we're adding a hacky workaround when deserialising the scene, it seems like we're missing the whole point of the Observer pattern and forcing it in. The Observer pattern is for communication in one-to-many scenarios, but it doesn't seem realistic that we're going to be communicating with multiple analytics observers.

Reply:
We had a discussed on this on the team and we came to the conclusion that until we have a method to achieve the same end in a way that is objectively better there's no real value in changing this.

Reply:
I thought we agreed to remove this or just return null if necessary ?

Reply:
Thought I had deleted it, the only use of this class was removed already, file deleted now